api/store/mongo/device.go

```diff
@@ -536,31 +536,26 @@ func (s *Store) DeviceChooser(ctx context.Context, tenantID string, chosen []str
 	return nil
 }
 
-// DeviceChooser updates devices with "accepted" status to "pending" for a given tenantID,
-// excluding devices with UIDs present in the "notIn" list.
 func (s *Store) DeviceUpdate(ctx context.Context, tenant string, uid models.UID, name *string, publicURL *bool) error {
-	session, err := s.db.Client().StartSession()
-	if err != nil {
-		return err
+	// Not deleting the device from the cache may cause issues when trying to retrieve the device after the update.
+	// TODO: Maybe we can standardize the key creation?
+	if err := s.cache.Delete(ctx, strings.Join([]string{"device", string(uid)}, "/")); err != nil {
+		logrus.Error(err)
 	}
```

Comment:
Delete the cache is an excellent idea, but I think that we should do it after a successful update only